deployment/docker/server/Dockerfile

```diff
@@ -25,8 +25,13 @@ RUN --mount=type=cache,target=/go/pkg \
 
 FROM alpine:3.19
 
+ARG TERRAFORM_VERSION="1.8.2"
```

Comment:
This should be defined as ENV var instead of an ARG to avoid continous rebuilds of all layers. Beside that it's IMHO a bad idea to bundle these tools. It increases the size of the image for everybody, also for people not using Terraform/OpenTofu/Pulimi integration adn just to avoid any kind of trouble with Hashicorp.

Reply:
@tboerger any ideas how to workaround this?

Reply:
create images:
* `semaphore-v*-terraform`
* `semaphore-v*-ansible`
* `semaphore-v*-all`
?

Reply:
Maybe add logic to download the binaries on demand when a task for it gets defined/starts?

Reply:
@tboerger yes, agree with you. We need to allow install required version of TF/Pulumi by demand. And I think Ansible version too.

But currently I will see only creating multiple images for different configurations:
```
semaphoreui/server:v2.10.1-ansible2.19 (base image)
semaphoreui/server:v2.10.1-ansible2.19-terraform1.8
semaphoreui/server:v2.10.1-ansible2.19-tofu1.7
semaphoreui/server:v2.10.1-ansible2.19-pulumi1.7
semaphoreui/server:v2.10.1-ansible2.19-tofu1.7-pulumi1.7
semaphoreui/server:v2.10.1-ansible2.19-tofu1.7-terraform1.8
semaphoreui/server:v2.10.1-ansible2.19-terraform1.8-tofu1.7-pulumi1.7
semaphoreui/server:v2.10.1-all
```

Reply:
Different ansible version can get installed via requirements.txt, IMHO you don't want to offer different versions of terraform/opentofu/pulumi, latest stable versions should be fine.

If the files for the providers are downloaded by need via semaphore you could even add an template option which defines and downloads the binaries.

No need to add that to the image at all.